en/middleware/micrortps.md

```diff
@@ -30,58 +32,84 @@ The main elements of the architecture are the client and agent processes shown i
+
+> **Caution** All new uORB messages which one adds to the Firmware side should have a respective RTPS ID set when one wants the data to be published and/or subscribed in the RTPS stream.
+
+```yaml
+rtps:
+  - msg: actuator_armed
+    id: 0
+  - msg: actuator_control
+    id: 1
+  - ...
+  - msg: airspeed
+    id: 5
+    send: true
+  - msg: battery_status
+    id: 6
+    send: true
+  - msg: camera_capture
+    id: 7
+  - msg: camera_trigger
+    id: 8
+    receive: true
 ```
 
-> **Caution** At time of writing (August 2017), only the small set of uORB topics listed above are included in our cmake files: **posix_sitl_default.cmake**, **nuttx_px4fmu-v4_default.cmake**, **posix_sdflight_default.cmake**. It is likely you will need to edit your *cmake* file and add additional uORB topics. In future we hope to define a larger standard set. 
+In the case of `px4_ros_com`, the `uorb_rtps_message_ids.yaml` is transformed in a way that the message names become PascalCased and there's a swap between the received and the sent messages. The logic behind this is that if a message is sent from the client side, then it's received on the agent side, and vice-versa. The naming of the message is indifferent on the client-agent communication, but it is critical on the ROS2 side, since the message naming has to follow the the PascalCase convention, which means that the generated IDL files also need to follow this same convention.
 
-For *manual code generation* the uORB topics that will be supported by the bridge are specified when you call **generate_microRTPS_bridge.py** (using the `-s`/`--send` and `-r`/`--receive` flags). See [Manual Generation of the Code](../middleware/micrortps_manual_code_generation.md) for more information.
+This simply means that, between agent and client, there are IDL files which are generated on the build process of PX4 Firmware and which are used only for communicate between each other, and from the agent side to the ROS2 RTPS middleware, there's another set of IDL files generated that are compatible with the ROS2 conventions. In the end, what really matters is that the generated IDL keep the same ID's and fields for each message.
 
 
 ## Client (PX4 Firmware)
 
-The *Client* source code is generated, compiled and built into the PX4 firmware as part of the normal build process. 
+The *Client* source code is generated, compiled and built into the PX4 firmware as part of the normal build process.
 
-To build and upload the firmware for NuttX/Pixhawk flight controllers:
+To build the firmware for NuttX/Pixhawk flight controllers:
 ```sh
-make px4fmu-v4_default upload
+make px4fmu-v4_rtps
```

Comment:
Is this change to the tooling in? 
This seems odd - does this mean that RTPS is not in firmware by default?

Reply:
It's not a default feature. You need to build the `micrortps_client`. It should become default soon.